sycl/include/sycl/ext/oneapi/properties/property_value.hpp

```diff
@@ -37,55 +38,43 @@ struct PropertyValueBase<T> : public detail::SingleNontypePropertyValueBase<T> {
 } // namespace detail
 
 template <typename PropertyT, typename... Ts>
-struct property_value : public detail::PropertyValueBase<Ts...> {
-  using key_t = PropertyT;
-};
+struct property_value : public detail::PropertyValueBase<Ts...> {};
 
 template <typename PropertyT, typename... A, typename... B>
-constexpr std::enable_if_t<detail::IsCompileTimeProperty<PropertyT>::value,
-                           bool>
-operator==(const property_value<PropertyT, A...> &,
-           const property_value<PropertyT, B...> &) {
+constexpr bool operator==(const property_value<PropertyT, A...> &,
+                          const property_value<PropertyT, B...> &) {
   return (std::is_same<A, B>::value && ...);
 }
 
 template <typename PropertyT, typename... A, typename... B>
-constexpr std::enable_if_t<detail::IsCompileTimeProperty<PropertyT>::value,
-                           bool>
-operator!=(const property_value<PropertyT, A...> &,
-           const property_value<PropertyT, B...> &) {
+constexpr bool operator!=(const property_value<PropertyT, A...> &,
+                          const property_value<PropertyT, B...> &) {
   return (!std::is_same<A, B>::value || ...);
 }
 
-template <typename V, typename = void> struct is_property_value {
```

Comment:
This is still part of the extension, so I don't think it should be removed. Even if it is to be removed, it would be more fitting to do in a separate patch.

Reply:
do you want me to also move the removal of tests for is_property_value and is_property_key into a follow up PR?